From c0a294a15710ebfef49c5491de767c390854bf32 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 24 May 2022 12:51:21 +0300
Subject: [PATCH] [SW-0] gaudi3: change security configuration sequence in recovery

Currently security configuration in recovery is perfomed after pcie link
and IATU configuration. This patch move the security configuration before
link up so that the security is configured before access to config space
and bar via link and iatu configuration.

Change-Id: I5e457468e1157f5fb80b13bc134704a6806f7548
---

diff --git a/arch/gaudi3/arc/recovery.c b/arch/gaudi3/arc/recovery.c
index 414a383..0436a3b 100644
--- a/arch/gaudi3/arc/recovery.c
+++ b/arch/gaudi3/arc/recovery.c
@@ -209,8 +209,6 @@
 
 	device_init();
 
-	security_rr_recovery_config();
-
 	module_handle_cmds(cmd_ops, ARRAY_SIZE(cmd_ops));
 }
 
@@ -286,6 +284,9 @@
 {
 	arcboot_rom_status_update(PROGRESS_ARCBOOT_IN_ROM_RECOVERY);
 
+	if (!arcboot_rom_global.die_id)
+		security_rr_recovery_config();
+
 	rom_pcie_recovery();
 	rom_d2d_link_init();
 	rom_recovery();
